var/spack/repos/builtin/packages/proj/package.py

```diff
@@ -128,10 +128,11 @@ def install_datum_grids(self):
 
 class CMakeBuilder(BaseBuilder, cmake.CMakeBuilder):
     def cmake_args(self):
+        shared_arg = "BUILD_SHARED_LIBS" if self.spec.satisfies("@7:") else "BUILD_LIBPROJ_SHARED"
         args = [
             self.define_from_variant("ENABLE_TIFF", "tiff"),
             self.define_from_variant("ENABLE_CURL", "curl"),
-            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+            self.define_from_variant(f"{shared_arg}", "shared"),
```

Comment:
```suggestion
            self.define_from_variant(shared_arg, "shared"),
```
It's already a string.

Reply:
Right